player/lua/stats.lua

```diff
@@ -721,114 +736,196 @@ local function pq_eotf(x)
+
     local scale = nil
     if not mp.get_property_native("fullscreen") then
         scale = mp.get_property_native("current-window-scale")
     end
-    append_resolution(s, {w=scaled_width, h=scaled_height, s=scale}, "Scaled Resolution:")
 
-    if mp.get_property_native("deinterlace") then
-        append_property(s, "deinterlace", {prefix="Deinterlacing:"})
+    local r = mp.get_property_native("video-target-params")
+    if not r then
+        local osd_dims = mp.get_property_native("osd-dimensions")
+        local scaled_width = osd_dims["w"] - osd_dims["ml"] - osd_dims["mr"]
+        local scaled_height = osd_dims["h"] - osd_dims["mt"] - osd_dims["mb"]
+        append_resolution(s, {w=scaled_width, h=scaled_height, s=scale},
+                          "Resolution:")
+        return
     end
 
-    append(s, r["pixelformat"], {prefix="Pixel Format:"})
-    if r["hw-pixelformat"] ~= nil then
-        append(s, r["hw-pixelformat"], {prefix_sep="[", nl="", indent=" ",
-                suffix="]"})
+    -- Add window scale
+    r["s"] = scale
+
+    append_img_params(s, r)
+    append_hdr(s, r, true)
+end
+
+
+local function add_video(s)
+    local r = mp.get_property_native("video-params")
+    local ro = mp.get_property_native("video-out-params")
+    -- in case of e.g. lavfi-complex there can be no input video, only output
+    if not r then
+        r = ro
+    end
+    if not r then
+        return
     end
-    append(s, r["colorlevels"], {prefix="Levels:", nl=""})
 
-    -- Group these together to save vertical space
-    append(s, r["colormatrix"], {prefix="Colormatrix:"})
-    append(s, r["primaries"], {prefix="Primaries:", nl=""})
-    append(s, r["gamma"], {prefix="Transfer:", nl=""})
+    local osd_dims = mp.get_property_native("osd-dimensions")
+    local scaled_width = osd_dims["w"] - osd_dims["ml"] - osd_dims["mr"]
+    local scaled_height = osd_dims["h"] - osd_dims["mt"] - osd_dims["mb"]
+
+    append(s, "", {prefix=o.nl .. o.nl .. "Video:", nl="", indent=""})
+    if append_property(s, "video-codec", {prefix_sep="", nl="", indent=""}) then
+        append_property(s, "hwdec-current", {prefix="HW:", nl="", indent=o.prefix_sep,
+                         no_prefix_markup=false, suffix=""}, {no=true, [""]=true})
+    end
+    local has_prefix = false
+    if append_property(s, "estimated-frame-number", {prefix="Frame:"}) then
+        append_property(s, "estimated-frame-count", {indent=" / ", nl="",
+                                                     prefix_sep=""})
+        has_prefix = true
+    end
+    local frame_info = mp.get_property_native("video-frame-info")
+    if frame_info and frame_info["picture-type"] then
+        local attrs = has_prefix and {prefix="(", suffix=")", indent=" ", nl="",
+                                      prefix_sep="", no_prefix_markup=true}
+                                  or {prefix="Picture Type:"}
+        append(s, frame_info["picture-type"], attrs)
+        has_prefix = true
+    end
```

Comment:
I also argue that displaying this is not useful due to:
> Note that if any of these are used on OSD, the information might be off by a few frames due to OSD redrawing and frame display being somewhat disconnected, and you might have to pause and force a redraw.

It will also be out of date due to refresh interval of the stats.

Reply:
I think it is obvious that stats are not updated every frame. And it is perfectly fine to pause if you want to inspect single frame in more details. Same applies to peak detection value.

Reply:
I generally frame step, which works really well.